backend/variables/builtin/array-find-index.js

```diff
@@ -0,0 +1,84 @@
+// Migration: done
+
+'use strict';
+
+const { OutputDataType, VariableCategory } = require("../../../shared/variable-constants");
+
+function getPropertyAtPath(obj, propertyPath) {
+    let data = obj;
+    const pathNodes = propertyPath.split(".");
+    for (let i = 0; i < pathNodes.length; i++) {
+        if (data == null) {
+            break;
+        }
+        let node = pathNodes[i];
+        // parse to int for array access
+        if (!isNaN(node)) {
+            node = parseInt(node);
+        }
+        data = data[node];
+    }
+    return data;
+}
+
+const model = {
+    definition: {
+        handle: "arrayFindIndex",
+        usage: "arrayFindIndex[jsonArray, matcher, propertyPath]",
+        description: "Finds a matching element in the array and returns it's index, or null if the element is absent",
+        examples: [
+            {
+                usage: 'arrayFindIndex["[\\"a\\",\\"b\\",\\"c\\"]", b]',
+                description: 'Returns 1 , the index of "b"'
+            },
+            {
+                usage: 'arrayFindIndex["[{\\"username\\": \\"alastor\\"},{\\"username\\": \\"ebiggz\\"}]", alastor, username]',
+                description: 'Returns 0, the index of the object where "username"="alastor"'
+            }
+        ],
+        categories: [VariableCategory.ADVANCED],
+        possibleDataOutput: [OutputDataType.TEXT, OutputDataType.NUMBER]
+    },
+    evaluator: (_, jsonArray, matcher, propertyPath = null) => {
+        if (jsonArray != null) {
+            if (matcher === undefined || matcher === "") {
+                return null;
+            }
+
+            try {
+                matcher = JSON.parse(matcher);
+            } catch (err) {
+                //fail silently
```

Comment:
return `null` instead of silently failing

Saves a bit of cycles needless-processing